Bottom: 74c916967de618742f5df336b6c6aca1160c120d
Top:    6a757a10982c7c5d0f1a3ce7624a449699e59106
Author: Jason Wang <jasowang@redhat.com>
Date:   2015-01-20 13:45:15 +0800

vhost: let vhost_signal() returns whether or not

Signed-off-by: Jason Wang <jasowang@redhat.com>


---

diff --git a/drivers/vhost/vhost.c b/drivers/vhost/vhost.c
index 08ff8a7..fadbe2f 100644
--- a/drivers/vhost/vhost.c
+++ b/drivers/vhost/vhost.c
@@ -1532,9 +1532,6 @@ void vhost_signal(struct vhost_dev *dev, struct vhost_virtqueue *vq)
 	int left;
 	ktime_t now;
 
-	if (!vq->call_ctx)
-		return;
-
 	if (can_coalesce) {
 		now = ktime_get();
 		left = vq->coalesce_usecs -
diff --git a/drivers/vhost/vhost.h b/drivers/vhost/vhost.h
index faa0dd7..e0ec89c 100644
--- a/drivers/vhost/vhost.h
+++ b/drivers/vhost/vhost.h
@@ -166,6 +166,8 @@ void vhost_add_used_and_signal(struct vhost_dev *, struct vhost_virtqueue *,
 void vhost_add_used_and_signal_n(struct vhost_dev *, struct vhost_virtqueue *,
 			       struct vring_used_elem *heads, unsigned count);
 void vhost_signal(struct vhost_dev *, struct vhost_virtqueue *);
+void vhost_check_coalesce_and_signal(struct vhost_dev *dev,
+				     struct vhost_virtqueue *vq);
 void vhost_disable_notify(struct vhost_dev *, struct vhost_virtqueue *);
 bool vhost_enable_notify(struct vhost_dev *, struct vhost_virtqueue *);
